tests/e2e/hci.go

```diff
@@ -488,4 +495,335 @@ var _ bool = ginkgo.Describe("hci", func() {
+
+		scParameters = map[string]string{}
+		scParameters[scParamStoragePolicyName] = remoteStoragePolicyName
+		scSpec := getVSphereStorageClassSpec(defaultNginxStorageClassName, scParameters, nil, "", "", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Create a sts with 1 replica")
+		sts := createCustomisedStatefulSets(ctx, client, namespace, false, replicas,
+			false, nil, false, true, "", "", sc, "")
+		defer func() {
+			ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+			fss.DeleteAllStatefulSets(ctx, client, namespace)
+		}()
+
+		nodes, err := client.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		pods := fss.GetPodList(ctx, client, sts)
+		targetWorkerName := pods.Items[0].Spec.NodeName
+
+		ginkgo.By("Isolate the host which has a k8s-worker with attached PVs in cluster1")
+		workerNode, err := client.CoreV1().Nodes().Get(ctx, targetWorkerName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		targetHost := e2eVSphere.getHostFromVMReference(ctx, getHostMoref4K8sNode(
+			ctx, client, workerNode))
+		finder := find.NewFinder(e2eVSphere.Client.Client, false)
+		var datacenters []string
+		cfg, err := getConfig()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		dcList := strings.Split(cfg.Global.Datacenters,
+			",")
+		for _, dc := range dcList {
+			dcName := strings.TrimSpace(dc)
+			if dcName != "" {
+				datacenters = append(datacenters, dcName)
+			}
+		}
+
+		for _, dc := range datacenters {
+			defDc, err := finder.Datacenter(ctx, dc)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			finder.SetDatacenter(defDc)
+			hosts, err := finder.HostSystemList(ctx, "*")
+			for _, host := range hosts {
+				if host.Reference().Reference().Value == targetHost.Value {
+					hostInfo := host.Common.InventoryPath
+					hostIpInfo := strings.Split(hostInfo, "/")
+					hostIp = hostIpInfo[len(hostIpInfo)-1]
+					break
+				}
+			}
+		}
+
+		framework.Logf("hostIp: %s", hostIp)
+
+		toggleNetworkFailureParallel([]string{hostIp}, true)
+		isTargetHostIsolated = true
+
+		defer func() {
+			if isTargetHostIsolated {
+				ginkgo.By("Re-integrate isolated host back into vsan cluster before terminating test")
+				toggleNetworkFailureParallel([]string{hostIp}, false)
+				isTargetHostIsolated = false
+			}
+		}()
+
+		ginkgo.By("Wait for 5-10 mins, verify that the k8s-worker is restarted and brought up on another host")
+		wait4AllK8sNodesToBeUp(nodes)
+		gomega.Expect(waitForAllNodes2BeReady(ctx, client)).To(gomega.Succeed())
+
+		time.Sleep(5 * time.Minute)
+
+		ginkgo.By("Scale up statefulset and scale down statefulset1 to 3 replicas")
```

Comment:
and scale down ? Do not see any scale down operation here.

Reply:
Will do it in next PR